pandas/_libs/lib.pyx

```diff
@@ -651,6 +651,13 @@ cpdef ndarray[object] ensure_string_array(
     cdef:
         Py_ssize_t i = 0, n = len(arr)
 
+    from pandas.core.dtypes.common import is_extension_array_dtype
+
+    if is_extension_array_dtype(arr):
```

Comment:
can we avoid doing this as it circularizs things, you can check if it has a `.to_numpy` method or maybe @jbrockmendel has a better way here.

Reply:
Ok, I've changed to use `hasattr(arr, "to_numpy")`.